File: docs/guides/java/src/test/java/com/algolia/saveObjectsChunks.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.File;
import java.util.List;
import java.util.Map;

public class saveObjectsChunks {

  public static void main(String[] args) throws Exception {
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY");) {
      JsonNode content = new ObjectMapper().readTree(new File("actors.json"));
      List<Map<String, Object>> records = new ObjectMapper().readerForListOf(Map.class).readValue(content);

      int chunkSize = 10000;

      for (var beginIndex = 0; beginIndex < records.size(); beginIndex += chunkSize) {
        List<Map<String, Object>> chunk = records.subList(beginIndex, Math.min(beginIndex + chunkSize, records.size()));
        client.saveObjects("<YOUR_INDEX_NAME>", chunk);
      }
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/saveObjectsChunks.kt
import java.io.File
import kotlinx.serialization.builtins.ListSerializer
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonObject

import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.transport.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*

suspend fun saveObjectsChunks() {
  val path = "/tmp/records.json"
  val json = try {
    File(path).readText()
  } catch (e: Exception) {
    throw RuntimeException("Failed to read file at $path", e)
  }
  val records: List<JsonObject> = Json.decodeFromString(ListSerializer(JsonObject.serializer()), json)

  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  records.chunked(10000).forEach { chunk ->
    try {
      client.saveObjects(indexName = "<YOUR_INDEX_NAME>", objects = chunk)
    } catch (exception: Exception) {
      println(exception.message)
    }
  }
}
